### PR TITLE
fix(utils): Make object $key and $exists properties non-enumerable

### PR DESCRIPTION
**Pull request by byrondover**
### Checklist

   - Issue number for this PR: #302
   - Docs included?: no
   - Test units included?: yes
   - e2e tests included?: no
   - In a clean directory, `npm install`, `npm run build`, and `npm test` run successfully? yes

### Description

Exact same pull request as #535, addresses rebase merge conflicts. All credit for original fix goes to @meDavid.

### CODE REVIEW

**Comment by byrondover** [comment]
I signed it!

**Comment by meDavid** [comment]
@byrondover Thanks you for finishing this

File: src/utils.ts [hunk 1]
```diff
@@ -57,10 +57,13 @@ export function unwrapMapFn (snapshot:firebase.database.DataSnapshot): AFUnwrapp
       $value: unwrapped
     };
   }
-  unwrapped.$key = snapshot.ref.key;
-  unwrapped.$exists = () => {
-    return snapshot.exists();
-  };
+  Object.defineProperty(unwrapped, '$key', {value: snapshot.ref.key, enumerable: false});
```
Comment (davideast) [inline src/utils.ts]: Format this line to the following:

```ts
Object.defineProperty(unwrapped, '$key', {
  value: snapshot.ref.key,
  enumerable: false
});
```

File: src/database/firebase_object_factory.spec.ts [hunk 2]
```diff
@@ -81,12 +81,12 @@ describe('FirebaseObjectFactory', () => {
 
 
     it('should emit unwrapped data by default', (done: any) => {
-      ref.set({ data: 'bar' }, () => {
+      const fixtureData = { data: 'bar' };
+      ref.set(fixtureData, () => {
         subscription = observable.subscribe(unwrapped => {
```
Comment (davideast) [inline src/database/firebase_object_factory.spec.ts]: The observable subscription shouldn't be needed here. Can you remove it and just use the inner code?

Comment (byrondover) [inline src/database/firebase_object_factory.spec.ts]: @davideast I'm not sure I follow. How can we assert the object properties after `ref.set()`, other than by subscribing to the object observable?

If it'll take too long to explain, please feel free to just tell me what to do here specifically, and I'm sure I'll figure it out as I implement the changes. :sunglasses:

File: src/utils.ts (hunk 1 above)
Comment (byrondover) [inline src/utils.ts]: Done! => ef09ba1

File: src/database/firebase_object_factory.spec.ts (hunk 2 above)
Comment (davideast) [inline src/database/firebase_object_factory.spec.ts]: Sorry, my brain was malfunctioning. I looked at this too early in the morning. I forgot that there's a `FirebaseObjectObservable` variable in the outer scope. All is good!

**Pull request merged**